packages/repository/src/model.ts

```diff
@@ -145,6 +149,63 @@ export class ModelDefinition {
     return this;
   }
 
+  /**
+   * Define a new "belongs to" relation.
+   * @param name - The name of the "belongs to" relation.
+   * @param definition - The definition for the "belongs to" relation.
+   */
+  belongsTo(
+    name: string,
+    definition: Partial<BelongsToDefinition> &
+      Pick<BelongsToDefinition, 'source' | 'target'>,
+  ): this {
+    const belongsToDefinition: BelongsToDefinition = {
+      ...definition,
+      name,
+      type: RelationType.belongsTo,
+      targetsMany: false,
+    };
+    return this.addRelation(belongsToDefinition);
+  }
+
+  /**
+   * Define a new "has one" relation.
+   * @param name - The name of the "has one" relation.
+   * @param definition - The definition of the "has one" relation.
+   */
+  hasOne(
+    name: string,
+    definition: Partial<HasOneDefinition> &
+      Pick<HasOneDefinition, 'source' | 'target'>,
```

Comment:
Similarly here, would the following work?

```suggestion
    definition: Omit<HasOneDefinition, 'name' | 'type' | 'targetsMany'>,
```